cargo-near/src/lib.rs

```diff
@@ -68,6 +76,56 @@ pub struct BuildCommand {
     /// Path to the `Cargo.toml` of the contract to build
     #[clap(long, parse(from_str), value_name = "PATH")]
     pub manifest_path: Option<Utf8PathBuf>,
+    /// Coloring: auto, always, never
+    #[clap(long, value_name = "WHEN")]
+    #[clap(default_value = "auto", possible_values = &["auto", "always", "never"])]
+    #[clap(hide_default_value = true, hide_possible_values = true)]
+    #[clap(parse(try_from_str = ColorPreference::from_str))]
+    pub color: ColorPreference,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum ColorPreference {
+    Always,
+    Never,
+}
+
+impl FromStr for ColorPreference {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "auto" => match env::var("NO_COLOR") {
+                Ok(v) if v != "0" => Ok(ColorPreference::Never),
+                _ => {
+                    if atty::is(atty::Stream::Stderr) {
+                        Ok(ColorPreference::Always)
+                    } else {
+                        Ok(ColorPreference::Never)
+                    }
+                }
+            },
+            "always" => Ok(ColorPreference::Always),
+            "never" => Ok(ColorPreference::Never),
+            _ => Err(format!("invalid value: {}", s)),
```

Comment:
Would be nice to indicate this is an "invalid color preference" or something like that, value is very generic

Reply:
Yep, sounds good to me.

Reply:
ah, interesting, wouldn't expect it to swallow the error. Optional for sure, but maybe saves some debug time of someone in the future :D